pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -451,11 +451,13 @@ def _tbptt_split_batch(self, batch: Any) -> List[Any]:
         Args:
             batch: the current batch to split
         """
-        splits = [batch]
-        if self.trainer.truncated_bptt_steps is not None:
-            model_ref = self.trainer.lightning_module
-            with self.trainer.profiler.profile("tbptt_split_batch"):
-                splits = model_ref.tbptt_split_batch(batch, self.trainer.truncated_bptt_steps)
+        tbptt_steps = self._truncated_bptt_steps()
+        if tbptt_steps == 0:
+            return [batch]
+
+        model_ref = self.trainer.lightning_module
+        with self.trainer.profiler.profile("tbptt_split_batch"):
+            splits = model_ref.tbptt_split_batch(batch, tbptt_steps)
```

Comment:
when the trainer deprecation of the tbtt argument ends, do you think we should still pass the steps to the hook or should the module directly use its property?

Reply:
I believe so. 

The step property today is serving 2 purposes:
- its presence & value indicates to the Trainer whether the user wants to use tbptt (ie change the loop behavior)
- its value is used for actually splitting batches

After the removal on the trainer side, one approach could be that we offer a TBPTTMixin, where `tbptt_split_batch` is a required method to implement. The user inherits this along with their LightningModule. 

```
class TBPTTModel(TBPTTMixin, LightningModule):
...
```

Then the Trainer could statically determine it should configure a corresponding tbptt_loop to run during training.

let's continue discussing on https://github.com/PyTorchLightning/pytorch-lightning/issues/8732 : )